### PR TITLE
Add azure-mgmt-synapse

### PR DESCRIPTION
**Pull request by janjagusch**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by janjagusch** [comment]
The source distribution on PyPI does not contain the license file: https://github.com/Azure/azure-sdk-for-python/issues/38633

**Comment by janjagusch** [comment]
> conda-forge-linting service

Seems like the conda-forge-linting service doesn't yet understand the rattler-build format. Can we still merge this? If not, I would have to revert https://github.com/conda-forge/staged-recipes/pull/28289/commits/689b9d4ef9bac5b27d00b950ac88326b8961d8b4.

File: recipes/azure-mgmt-synapse/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+schema_version: 1
+
```
Comment (xhochy) [inline recipes/azure-mgmt-synapse/recipe.yaml:2]: I have not seen this in other PRs
```suggestion
```

File: recipes/azure-mgmt-synapse/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  version: 2.0.0
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  - url: https://pypi.org/packages/source/${{ name[0] }}/${{ name }}/azure-mgmt-synapse-${{ version }}.zip
+    sha256: bec6bdfaeb55b4fdd159f2055e8875bf50a720bb0fce80a816e92a2359b898c8
+  - url: https://raw.githubusercontent.com/Azure/azure-sdk-for-python/70f714969c6311d02f41632785e2dc6d3595a826/LICENSE
```
Comment (janjagusch) [inline recipes/azure-mgmt-synapse/recipe.yaml:14]: That should be the correct license file.

```suggestion
  - url: https://raw.githubusercontent.com/Azure/azure-sdk-for-python/50471e7f07b430ab0bbc1984cd2fc102d908de41/sdk/synapse/azure-mgmt-synapse/LICENSE
```

File: recipes/azure-mgmt-synapse/recipe.yaml (hunk 1 above)
Comment (janjagusch) [reply recipes/azure-mgmt-synapse/recipe.yaml:2]: This was auto-generated by `greyskull pypi --use-v1-format`.

**Pull request merged**